Automattic/simplenote-ios
# Extended Debugging

### Details:
This PR adds an "Extended Diagnostics Mode" to the Authentication Interface.

cc @SylvesterWilmott What do you think? (Hackweek Project!!)

@etoledom Eduardoo!!! Can I bug you with a fun (easter egg) PR?

Thank youuu!!


### Test
1. Fresh Install Simplenote
2. **Triple Tap** over the Simplenote Logo

- [x] Verify that an alert saying `Debug Enabled` shows up
- [x] Verify that triple-tapping again triggers a `Debug Disabled` alert

### Test: Diagnostics
1. Fresh Install Simplenote
2. **Triple Tap** over the Simplenote Logo
3. Press over **Log In > Log In with Email**
4. Set your phone in Airplane Mode
5. Enter any random credentials and **press Login**

- [x] Verify a new UI shows up, with extra debug details
- [x] Verify that pressing `Share` shows the standard System's Share Sheet
- [x] Verify that pressing `Done` dismisses the Diagnostics UI

### Release
These changes do not require release notes.

---

### Screenshots

<img width="300" src="https://user-images.githubusercontent.com/1195260/93524871-889a6e00-f90b-11ea-853e-34d97117fe34.png"> <img width="300" src="https://user-images.githubusercontent.com/1195260/93525002-c0a1b100-f90b-11ea-877b-e6ec4733c084.PNG">
